### PR TITLE
refactor(runtime): hide internals from public-facing API, export tokio

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

*finally a well isolated PR...*

- Move internals into hidden module, which hides them from the cargo doc page (docs.rs).
- Re-export tokio, allows user code to not have tokio in Cargo.toml, unless they need it for something ofc :)

### CODE REVIEW

**Review by oddgrd — APPROVED** [review]

**Comment by oddgrd** [comment]
Cool! If the user wants a different version of tokio, cargo should use the one they specify, right (as long as it's not a major version change, but that isn't planned for tokio)?

**Comment by jonaro00** [comment]
Yes, the min version remains what we have in our Cargo.toml (1.22), and creating or bumping the lock file will choose the latest compatible one.

**Review by orhun — APPROVED** [review]

File: codegen/src/shuttle_main/mod.rs [hunk 1]
```diff
@@ -14,9 +14,13 @@ pub(crate) fn r#impl(_attr: TokenStream, item: TokenStream) -> TokenStream {
     let loader = Loader::from_item_fn(&mut fn_decl);
 
     quote! {
-        #[tokio::main]
-        async fn main() {
-            shuttle_runtime::start(loader).await;
+        fn main() {
+            use ::shuttle_runtime::tokio;
+            #[tokio::main]
+            async fn main() {
```
Comment (chesedo) [inline codegen/src/shuttle_main/mod.rs:20]: Why is there a `main()` inside `main`?

File: services/shuttle-poise/src/lib.rs [hunk 2]
```diff
@@ -22,7 +21,7 @@
 //!     // Get the discord token set in `Secrets.toml`
 //!     let discord_token = secret_store
 //!         .get("DISCORD_TOKEN")
-//!         .context("'DISCORD_TOKEN' was not found")?;
```
Comment (chesedo) [inline services/shuttle-poise/src/lib.rs:25]: Why this change?

**Review by chesedo — COMMENTED** [review]
Looking good. It would w great to run a `cargo check` on all our examples in case we missed anything with the review.

File: codegen/src/shuttle_main/mod.rs (hunk 1 above)
Comment (jonaro00) [reply codegen/src/shuttle_main/mod.rs:20]: To scope off our tokio import and not leak it to the end user's main.rs

File: services/shuttle-poise/src/lib.rs (hunk 2 above)
Comment (jonaro00) [reply services/shuttle-poise/src/lib.rs:25]: It depends on runtime re-exporting anyhow, which is a bit unnecessary

Comment (chesedo) [reply services/shuttle-poise/src/lib.rs:25]: It just seems that the old error would have been a better practise

File: codegen/src/shuttle_main/mod.rs (hunk 1 above)
Comment (chesedo) [reply codegen/src/shuttle_main/mod.rs:20]: Maybe we should not call the `tokio::main` macro then and just expand it ourselves.
https://docs.rs/tokio/latest/tokio/attr.main.html#using-the-multi-thread-runtime

Comment (chesedo) [reply codegen/src/shuttle_main/mod.rs:20]: I'm trying to reduce the confusion here for the future

File: codegen/src/shuttle_main/mod.rs [hunk 3]
```diff
@@ -14,9 +14,15 @@ pub(crate) fn r#impl(_attr: TokenStream, item: TokenStream) -> TokenStream {
     let loader = Loader::from_item_fn(&mut fn_decl);
 
     quote! {
-        #[tokio::main]
-        async fn main() {
-            shuttle_runtime::start(loader).await;
+        fn main() {
+            // manual expansion of #[tokio::main]
+            ::shuttle_runtime::tokio::runtime::Builder::new_multi_thread()
```
Comment (chesedo) [inline codegen/src/shuttle_main/mod.rs:19]: Can we turn off some tokio defaults because of this? I'm thinking of turning off the `macros` feature to reduce the dependencies on users.

**Review by chesedo — APPROVED** [review]
We can merge if the dependencies improvement is not possible 😄

File: codegen/src/shuttle_main/mod.rs (hunk 3 above)
Comment (chesedo) [reply codegen/src/shuttle_main/mod.rs:19]: We can also do it in a separate PR

Comment (jonaro00) [reply codegen/src/shuttle_main/mod.rs:19]: It seems like other dependencies are enabling it. It's a messy task.
